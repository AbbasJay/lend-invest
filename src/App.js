const App = () => {
  return <div className="App">Lendinvest test</div>;
};

export default App;
