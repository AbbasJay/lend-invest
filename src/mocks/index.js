export const mockInitialState = () => ({
  currentLoans: {
    loans: [
      {
        id: "1",
        available: "8,250",
      },
      {
        id: "2",
        available: "14,250",
      },
      {
        id: "3",
        available: "2,250",
      },
    ],
  },
});
