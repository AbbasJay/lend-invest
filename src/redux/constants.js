const prefix = "currentLoans/";

export const SET_CURRENT_LOANS = `${prefix}SET_CURRENT_LOANS`;

export const SET_SELECTED_LOAN = `${prefix}SET_SELECTED_LOAN`;

export const SHOW_INVEST_MODAL = `${prefix}SHOW_INVEST_MODAL`;
export const HIDE_INVEST_MODAL = `${prefix}HIDE_INVEST_MODAL`;

export const SET_INVESTED_AMOUNT = `${prefix}SET_INVESTED_AMOUNT`;
