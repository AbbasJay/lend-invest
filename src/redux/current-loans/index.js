const currentLoansReducer = (state = []) => {
  return state;
};

export default currentLoansReducer;
