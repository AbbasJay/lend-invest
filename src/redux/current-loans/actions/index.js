import * as constants from "../constants";

export const setCurrentLoans = (currentLoans) => ({
  type: constants.SET_CURRENT_LOANS,
  payload: { currentLoans },
});
