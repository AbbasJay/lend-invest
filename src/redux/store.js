import { createStore, combineReducers } from "redux";
import { composeWithDevTools } from "redux-devtools-extension";

import currentLoansReducer from "./current-loans";

const composeEnhancer =
  window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || composeWithDevTools();

export default createStore(
  combineReducers({
    currentLoans: currentLoansReducer,
  }),
  composeEnhancer()
);
