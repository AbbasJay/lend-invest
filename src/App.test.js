import { Provider } from "react-redux";
import renderer from "react-test-renderer";

import { mount } from "enzyme";

import { setCurrentLoans } from "./redux/actions";

import { mockInitialState } from "./mocks";

import store from "./redux/store";
import App from "./App";

const mockDispatch = jest.spyOn(store, "dispatch");

const mockCurrentLoans = mockInitialState.currentLoans.loans;

const renderedComponent = () => {
  return mount(
    <Provider store={store}>
      <App />
    </Provider>
  );
};

describe("App", () => {
  beforeEach(() => {
    jest.resetAllMocks();
  });

  it("renders correctly to snapshot", () => {
    const component = renderedComponent();
    const snapshot = renderer.create(component).toJSON();

    expect(snapshot).toMatchSnapshot();
  });

  it("should render component", () => {
    const component = renderedComponent();

    expect(component.length).toBe(1);
  });

  it("should dispatch correct action", () => {
    store.dispatch(setCurrentLoans(mockCurrentLoans));

    expect(mockDispatch).toHaveBeenCalledWith(
      setCurrentLoans(mockCurrentLoans)
    );
  });
});
