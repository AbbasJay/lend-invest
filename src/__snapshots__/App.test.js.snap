// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly to snapshot 1`] = `
<div>
  loading...
</div>
`;
